client/src/features/Main/Rush.tsx

```diff
@@ -1,49 +1,55 @@
+import RushEvent, { TotalRushEventsProps } from "@/components/RushEvent";
+import { RUSH_EVENT_DATA } from "@/constants/Main/rushEventData.ts";
 import { Section } from "@/features/Main/Section.tsx";
+import useFetch from "@/hooks/useFetch.ts";
+import { GetTotalRushEventsResponse } from "@/types/rushApi.ts";
 import { SectionKeyProps } from "@/types/sections.ts";
-
-// TODO: API로 대체될 데이터
-export const rushEventData: RushEventProps[] = [
-    {
-        id: 1,
-        date: "2024-07-28 09:00:00.000000",
-        image: "/assets/main/rush/prize-1.png",
-        prizeName: "영화 예매권",
-    },
-    {
-        id: 2,
-        date: "2024-07-29 09:00:00.000000",
-        image: "/assets/main/rush/prize-2.png",
-        prizeName: "야구 관람권",
-    },
-    {
-        id: 3,
-        date: "2024-07-30 09:00:00.000000",
-        image: "/assets/main/rush/prize-3.jpg",
-        prizeName: "올리브영 상품권",
-    },
-    {
-        id: 4,
-        date: "2024-07-31 09:00:00.000000",
-        image: "/assets/main/rush/prize-4.jpeg",
-        prizeName: "쿠팡 기프트카드",
-    },
-    {
-        id: 5,
-        date: "2024-08-01 09:00:00.000000",
-        image: "/assets/main/rush/prize-5.jpg",
-        prizeName: "배달의민족 기프트카드",
-    },
-    {
-        id: 6,
-        date: "2024-08-02 09:00:00.000000",
-        image: "/assets/main/rush/prize-6.jpg",
-        prizeName: "BBQ 기프트카드",
-    },
-];
+import { formatEventDateRangeWithDot } from "@/utils/formatDate.ts";
+import { getMsTime } from "@/utils/getMsTime.ts";
 
 function Rush({ id }: SectionKeyProps) {
+    const [rushEvents, setRushEvents] = useState<TotalRushEventsProps[]>([]);
+    const [startDateTime, setStartDateTime] = useState<string>("");
+    const [endDateTime, setEndDateTime] = useState<string>("");
+
+    const {
+        data: rushData,
+        isSuccess: isSuccessRush,
+        fetchData: getRush,
+    } = useFetch<GetTotalRushEventsResponse>(() => RushAPI.getRush());
+
+    useEffect(() => {
+        getRush();
+    }, []);
+
+    useEffect(() => {
+        if (isSuccessRush && rushData) {
+            const serverDateTime = getMsTime(rushData.serverTime);
+
+            setStartDateTime(rushData.eventStartDate);
+            setEndDateTime(rushData.eventEndDate);
+
+            const events = rushData.events.map((event, idx) => {
+                const rushEvent = RUSH_EVENT_DATA[idx];
+                const eventEndTime = getMsTime(event.endDateTime);
+
+                return {
+                    id: event.rushEventId,
+                    date: event.startDateTime,
+                    image: rushEvent?.image || "",
+                    prizeName: rushEvent?.prizeName || "",
+                    isPastEvent: serverDateTime > eventEndTime,
+                    isTodayEvent:
+                        event.rushEventId === rushData.todayEventId &&
+                        serverDateTime <= eventEndTime,
+                };
+            });
+            setRushEvents(events);
```

Comment:
요 데이터들 굳이 다른 state로 관리 안 하고 바로 가져다 써도 되지 않을까용?

```suggestion
           const { eventStartDate, eventEndDate, todayEventId, rushEventId, image, prizeName } = rushData;
            const events = useMemo(() => rushData.events.map((event, idx) => {
                const rushEvent = RUSH_EVENT_DATA[idx];
                const eventEndTime = getMsTime(event.endDateTime);

                return {
                    id: event.rushEventId,
                    date: event.startDateTime,
                    image: rushEvent?.image || "",
                    prizeName: rushEvent?.prizeName || "",
                    isPastEvent: serverDateTime > eventEndTime,
                    isTodayEvent:
                        event.rushEventId === rushData.todayEventId &&
                        serverDateTime <= eventEndTime,
                };
            }), [rushData])
```

어차피 useFetch가 반환하는 값이 state니까 따로 state 안 만들고 약간 요런식으로 해서 useEffect 내부가 아니라 컴포넌트 내부에서 바로 사용해도 괜찮지 않을까 싶어서요,,!

Reply:
반영했습니다! 유사하게 구현한 Main/Headline.tsx, Main/Lottery.tsx 부분에도 구조분해할당해서 바로 내부에서 호출하도록 변경했습니다!